src/main/java/com/project/controller/UserController.java

```diff
@@ -0,0 +1,62 @@
+package com.project.controller;
+
+import com.project.exception.UserException;
+import com.project.exception.UserNotFoundException;
+import com.project.model.Application;
+import com.project.model.request.UserCommentRequest;
+import com.project.model.request.UserRequest;
+import com.project.model.response.ApplicationResponse;
+import com.project.model.response.UserResponse;
+import com.project.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@CrossOrigin(origins = "*")
+@RequestMapping("api/users")
+public class UserController {
+
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<UserResponse>> fetchUsers() {
+        return new ResponseEntity<>(userService.getAll(), HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    ResponseEntity<UserResponse> fetchUser(@PathVariable Long id) throws UserNotFoundException {
+        return new ResponseEntity<>(new UserResponse(userService.getById(id)), HttpStatus.OK);
+    }
+
+    @PostMapping("/application")
+    public ResponseEntity<ApplicationResponse> fetchUserApplication(@RequestBody UserCommentRequest email) throws UserNotFoundException {
```

Comment:
Why this service fetches application in a user controller also why it has a `UserCommentRequest` as request but not the Comment is fetched from the service instead it fetches the Application?

Reply:
renamed `UserCommentRequest` to `UserApplicationRequest` to better reflect what it is used for. As for why does this method fetch an application in the user controller, it is because it uses the same request mapping path.